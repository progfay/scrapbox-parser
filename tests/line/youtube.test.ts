describe("youtube", () => {
  it("Youtube URL", () => {
    expect(
      "[https://youtube.com/watch?v=Sv3xVOs7_No]"
    ).toMatchSnapshotWhenParsing({
      hasTitle: false,
    });
  });

  it("Youtube short URL", () => {
    expect("[https://youtu.be/Sv3xVOs7_No]").toMatchSnapshotWhenParsing({
      hasTitle: false,
    });
  });

  it("Youtube watch list URL", () => {
    expect(
      "[https://www.youtube.com/playlist?list=PLmoRDY8IgE2OTDG7rVwVLd-szijCAZyZC]"
    ).toMatchSnapshotWhenParsing({
      hasTitle: false,
    });
  });
});
