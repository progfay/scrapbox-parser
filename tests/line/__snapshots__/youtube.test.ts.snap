// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`youtube Youtube URL: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "params": Object {
          "v": "Sv3xVOs7_No",
        },
        "raw": "[https://youtube.com/watch?v=Sv3xVOs7_No]",
        "src": "https://youtube.com/watch?v=Sv3xVOs7_No",
        "type": "youtube",
        "videoId": "Sv3xVOs7_No",
      },
    ],
    "type": "line",
  },
]
`;

exports[`youtube Youtube short URL: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "params": Object {},
        "raw": "[https://youtu.be/Sv3xVOs7_No]",
        "src": "https://youtu.be/Sv3xVOs7_No",
        "type": "youtube",
        "videoId": "Sv3xVOs7_No",
      },
    ],
    "type": "line",
  },
]
`;

exports[`youtube Youtube watch list URL: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "listId": "PLmoRDY8IgE2OTDG7rVwVLd-szijCAZyZC",
        "params": Object {
          "list": "PLmoRDY8IgE2OTDG7rVwVLd-szijCAZyZC",
        },
        "raw": "[https://www.youtube.com/playlist?list=PLmoRDY8IgE2OTDG7rVwVLd-szijCAZyZC]",
        "src": "https://www.youtube.com/playlist?list=PLmoRDY8IgE2OTDG7rVwVLd-szijCAZyZC",
        "type": "youtube",
      },
    ],
    "type": "line",
  },
]
`;
